aikau/src/main/resources/alfresco/renderers/Thumbnail.js

```diff
@@ -499,6 +619,161 @@ define(["dojo/_base/declare",
+            if (dimensions.h)
+            {
+               thumbnailHeight = parseInt(dimensions.h, 10);
+            }
+
+            // Get the margin and border (in order to calculate the appropriate
+            // inner image size)...
+            var margin = 0;
+            if (dimensions.margin)
+            {
+               margin = parseInt(dimensions.margin, 10);
+            }
+
+            // Calcuate the image dimensions...
+            var borderThickness = parseInt(domStyle.get(this.imgNode, "borderWidth"), 10);
+            this.imageNodeHeight = thumbnailHeight - (margin + borderThickness);
+            this.imageNodeWidth = thumbnailHeight - (margin + borderThickness);
+            
+            // Update the thumbnail nodes...
+            domStyle.set(this.thumbnailNode, {
+               "width": thumbnailWidth + "px",
+               "height": thumbnailHeight + "px"
+            });
+            domStyle.set(this.frameNode, {
+               "width": thumbnailWidth + "px",
+               "height": thumbnailHeight + "px",
+               "lineHeight": thumbnailHeight + "px"
+            });
+            domStyle.set(this.imgNode, "margin", margin + "px");
+
+            if (this.maintainAspectRatio)
+            {
+               domStyle.set(this.imgNode, {
+                  "maxWidth": this.imageNodeWidth + "px",
+                  "maxHeight": this.imageNodeHeight + "px"
+               });
+            }
+            else if (!this.cropToFit)
+            {
+               domStyle.set(this.imgNode, {
+                  "width": this.imageNodeWidth + "px",
+                  "minHeight": this.imageNodeWidth + "px",
+                  "maxHeight": "none",
+                  "maxWidth": "none"
+               });
+            }
+
+            if (this.cropToFit && this.naturalImageWidth && this.naturalHeight)
+            {
+               this.alfLog("info", "Cropping on resize");
+               this.cropImage();
+            }
+         }
+      },
+
+      /**
+       * This function is used to configure the thumbnail DOM model so that the image is cropped and
+       * centered within the thumbnail. It relies on 
+       * [getNaturalImageSize]{@link module:alfresco/renderers/Thumbnail#getNaturalImageSize} 
+       * having been called to establish the orientation of the image. Therefore it is important that
+       * this is only called when the natural image dimensions have been established.
+       *
+       * @instance
+       * @since 1.0.40
+       */
+      cropImage: function alfresco_renderers_Thumbnail__cropImage() {
+         domStyle.set(this.frameNode, {
+            position: "relative",
+            overflow: "hidden",
+            textAlign: "left"
+         });
+         if (this.naturalImageHeight > this.naturalImageWidth)
+         {
+            var vOffset = ((this.imageNodeWidth / (this.naturalImageWidth / this.naturalImageHeight)) - this.imageNodeWidth) / 2;
```

Comment:
Might be nice to have `this.naturalImageWidth / this.naturalImageHeight` as a `naturalAspectRatio` variable, and then add an additional `isPortrait = naturalAspectRatio < 1` variable, making the condition above easier to read (i.e. `if (isPortrait) {`) as well as reducing duplicated calculations in the offset calculations.